menas/src/main/scala/za/co/absa/enceladus/menas/services/SchemaRegistryService.scala

```diff
@@ -69,8 +120,34 @@ class SchemaRegistryService @Autowired()() {
    */
   def loadSchemaByUrl(remoteUrl: String): SchemaResponse = {
     try {
+      if (keyManagerFactory.isEmpty) {
+        logger.warn("keyManagerFactory is not defined, secure schema registry integration may not function properly. " +
+          s"Set ${SecureConfig.Keys.javaxNetSslKeyStore} and ${SecureConfig.Keys.javaxNetSslKeyStorePassword} to fix the issue.")
+      }
+      if (trustManagerFactory.isEmpty) {
+        logger.warn("trustManagerFactory is not defined, secure schema registry integration may not function properly. " +
+          s"Set ${SecureConfig.Keys.javaxNetSslTrustStore} and ${SecureConfig.Keys.javaxNetSslTrustStorePassword} to fix the issue.")
+      }
+
+      val ctx = SSLContext.getInstance(defaultSslContextProtocol)
+      ctx.init(
+        keyManagerFactory.map(_.getKeyManagers).orNull,
+        trustManagerFactory.map(_.getTrustManagers).orNull,
+        null // scalastyle:ignore null - Java API
+      )
+
       val url = new URL(remoteUrl)
       val connection = url.openConnection()
+
+      if (remoteUrl.toLowerCase.startsWith("https")) {
+        val httpsConnection: HttpsURLConnection = connection.asInstanceOf[HttpsURLConnection]
+        httpsConnection.setSSLSocketFactory(ctx.getSocketFactory)
+      } else {
+        logger.warn(s"Connection to schema registry at $remoteUrl is not secure and trust/key stores are not used." +
```

Comment:
Ditto on adding a suppression option for the warning.

Reply:
Well, here it only applies to 'https', so I believe the warning is quite correct. Don't you think? But still, I can probably let it drive with the same flag as above.